lib/resty/acme/storage/redis.lua

```diff
@@ -55,8 +57,31 @@ local function op(self, op, ...)
   return ok, err
 end
 
+local function add_namespace(namespace, key)
+  if namespace == "" then
+    return key
+  else
+    -- <namespace>::<real_key>
+    return fmt("%s::%s", namespace, key)
+  end
+end
+
+local function remove_namespace(namespace, keys)
+  if namespace == "" then
+    return keys
+  else
+    -- <namespace>::<real_key>
+    local start = #namespace + 2 + 1
+    for k, v in ipairs(keys) do
+      keys[k] = v:sub(start)
```

Comment:
apart from removing the namespace, let's also double check if the removed portion is indeed the namespace;
if not, just return as is and log in error log.